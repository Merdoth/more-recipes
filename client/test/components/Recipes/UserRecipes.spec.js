import React from 'react';
import { shallow } from 'enzyme';
import thunk from 'redux-thunk';
import expect from 'expect';
import configureMockStore from 'redux-mock-store';
import ConnectedUserRecipes, { UserRecipes } from
  '../../../components/Recipes/UserRecipes.jsx';
import mockData from '../../__mocks__/mockData';


const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
/* global jest */

const setup = () => {
  const props = {
    recipes: [
      {
        id: 1,
        userId: 1,
        recipeName: 'Sweet cake',
        description: 'sweet',
        ingredients: 'floor, sugar, butter, salt',
        preparation: 'Mix and bake',
        image: '/img/toast_qaxxcy.jpg',
        views: 0,
        upVotes: 0,
        downVotes: 0,
        createdAt: '2018-02-13T22:14:54.283Z',
        updatedAt: '2018-02-18T15:23:59.106Z',
      }
    ],
    pagination: () => { },
    getUserRecipes: () => { }
  };
  return shallow(<UserRecipes {...props} />);
};

describe('UserRecipes Component snapshot', () => {
  it('it should render the right amount of elements', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Connected UserRecipes component', () => {
  it('tests that the component successfully rendered', () => {
    const { recipes } = mockData;
    const store = mockStore({
      recipesReducer: {
        recipesFound: recipes.recipesFound
      }
    });
    const wrapper = shallow(<ConnectedUserRecipes store={store} />);
    expect(wrapper.length).toBe(1);
  });
});
