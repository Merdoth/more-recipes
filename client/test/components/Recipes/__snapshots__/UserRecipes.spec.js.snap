// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserRecipes Component snapshot it should render the right amount of elements 1`] = `
<div>
  <div
    className="container manage"
  >
    <div
      className="recipe-header"
    >
      <h2>
        Recipes
      </h2>
      <p>
        Feel free to manage your own account
      </p>
    </div>
  </div>
  <div
    className="recipe-wrapper"
  >
    <div
      className="container top"
    >
      <div
        className="row"
      >
        <RecipeCard
          key="recipes-1"
          recipeList={
            Object {
              "createdAt": "2018-02-13T22:14:54.283Z",
              "description": "sweet",
              "downVotes": 0,
              "id": 1,
              "image": "/img/toast_qaxxcy.jpg",
              "ingredients": "floor, sugar, butter, salt",
              "preparation": "Mix and bake",
              "recipeName": "Sweet cake",
              "upVotes": 0,
              "updatedAt": "2018-02-18T15:23:59.106Z",
              "userId": 1,
              "views": 0,
            }
          }
        />
      </div>
      <div
        className="row pagination"
      >
        <PaginationBoxView
          activeClassName="active"
          breakClassName="break-me"
          breakLabel="..."
          containerClassName="pagination"
          disableInitialCallback={false}
          disabledClassName="disabled"
          marginPagesDisplayed={2}
          nextClassName="next"
          nextLabel="next"
          onPageChange={[Function]}
          pageCount={10}
          pageRangeDisplayed={5}
          previousClassName="previous"
          previousLabel="previous"
          subContainerClassName="pages pagination"
        />
      </div>
    </div>
  </div>
</div>
`;
